content/blog/2020/01/2020-01-23-ui-plugins.adoc

```diff
@@ -0,0 +1,708 @@
+in the warnings plugin to see how such a table can be decorated.
+
+[#tables-dynamic]
+==== Tables with dynamic model based content
+
+While static HTML tables are easy to implement, they have several limitations. So it makes sense to follow a more
+sophisticating approach. Typically, tables in user interfaces are defined by using a corresponding table (and row) model.
+Java Swing successfully provides such a
+https://docs.oracle.com/javase/tutorial/uiswing/components/table.html[table model] concept since the early days of Java.
+I adapted these concepts for Jenkins and DataTables as well. In order to create a table in a Jenkins view a plugin
+needs to provide a table model class, that provides the following information:
+
+- the ID of the table (since there might be several tables in the view)
+- the model of the columns (i.e., the number, type, and header labels of the columns)
+- the content of the table (i.e. the individual row objects)
+
+You will find an example of such a table in the Forensics plugin: here a table lists
+the files in your Git repository combined with the corresponding commit statistics (number of authors,
+number of commits, last modification, first commit). A screenshot of that table is shown in <<img-table>>.
+
+.Dynamic Table in the Forensics plugin
+[#img-table]
+image::table.png[Table example]
+
+In order to create such a table in Jenkins, you need to create a table model class that derives from `TableModel`.
+In <<forensics-table-model>> a diagram of the corresponding classes in the Forensics plugin is shown.
+
+[#forensics-table-model]
+.Table model of the Forensics plugin
+image::table-model.png[Tabel model]
+
+===== Table column model
+
+This first thing a table model class defines is a model of the available columns by creating corresponding
+ `TableColumn` instances. For each column you need to specify a header label and the name of the bean property that
+should be shown in the corresponding column (the row elements are actually Java beans: each column will
+show one distinct property of such a bean, see next section). You can
+use any of the https://datatables.net/reference/option/columns.type[supported column types] by simply providing a
+`String` or `Integer` based column.
+
+===== Table rows content
+
+Additionally, a table model class provides the content of the rows. This `getRows()` method
+will be invoked asynchronously using an Ajax call. Typically, this method simply returns a list of Java Bean instances,
+that provide the properties of each column (see previous section). These objects will be converted automatically
+to an array of JSON objects, the basic data structure required for the DataTables API.
+You will find a fully working example table model
+implementation in the Git repository of the forensics plugin in the class
+https://github.com/jenkinsci/forensics-api-plugin/blob/plugin-util/src/main/java/io/jenkins/plugins/forensics/miner/ForensicsTableModel.java[ForensicsTableModel].
```

Comment:
Since the plugin-util branch no longer exists, point to the master branch instead.
```suggestion
https://github.com/jenkinsci/forensics-api-plugin/blob/master/src/main/java/io/jenkins/plugins/forensics/miner/ForensicsTableModel.java[ForensicsTableModel].
```